packages/vike-react-apollo/src/withFallback.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { ComponentType, ReactNode, Suspense } from 'react'
+import { ErrorBoundary, FallbackProps } from 'react-error-boundary'
+import { ApolloConsumer } from '@apollo/client/index.js'
+type RetryOptions = { retryQuery?: boolean }
+type RetryFn = (options?: RetryOptions) => void
+
+type ErrorFallbackProps = {
+  error: { message: string } & Record<string, unknown>
+  retry: RetryFn
+}
+
+type Loading<T> = ComponentType<T> | ReactNode
+type Error<T> = ComponentType<T & ErrorFallbackProps> | ReactNode
+
+type WithFallbackOptions<T> = {
+  Loading?: Loading<T>
+  Error?: Error<T>
+}
+
+export function withFallback<T extends object = Record<string, never>>(
+  Component: ComponentType<T>,
+  options?: WithFallbackOptions<T>
```

Comment:
Actually, instead of having an object, how about we support `withFalllback(Component, null, Error)`?

Reply:
It's supported already. Should we remove support for the object and only keep `withFalllback(Component, null, Error)`?

Reply:
> It's supported already. Should we remove support for the object and only keep `withFalllback(Component, null, Error)`?

I'd be :+1: for removing support for passing an object. (Option 1.)

Alternatively (option 2), we could support only passing an object for defining `Error`:

```js
// Not supported
withFalllback(Component, null, Error)

// Supported
withFalllback(Component, { Error })
withFalllback(Component, Loading)
withFalllback(Component, { Loading, Error })
```

But, so far, I prefer option 1.

I'm thinking that supporting both option 1 and option 2 adds more noise than adds benefit.